src/atldld/utils.py

```diff
@@ -95,60 +96,44 @@ def get_image(image_id, folder=None, expression=False, downsample=0):
     Raises
     ------
     ValueError
-        If request to download the image failed.
+        If the image has a wrong format (determined by the dtype).
 
     References
     ----------
     [1] `AllenSDK API: ImageDownloadApi <https://allensdk.readthedocs.io/
     en/latest/allensdk.api.queries.image_download_api.html#allensdk.api.
     queries.image_download_api.ImageDownloadApi>`_
-
     """
     folder = folder or CACHE_FOLDER
-    folder = str(
-        folder
-    )  # this should guarantee that also LocalPath works (pytest uses it)
-
-    # Check the folder exists
-    if not os.path.exists(folder):
-        os.makedirs(folder)
-
-    # Create full path
-    additional_specifier = "_expression" if expression else ""
-    additional_specifier += f"_{downsample}"
-    path = "{}{}{}.jpg".format(folder, image_id, additional_specifier)
-
-    # Check image exists
-    if os.path.exists(path):
-        img = plt.imread(path)
-
-        if not img.dtype == np.uint8:
-            raise ValueError("The dtype needs to be uint8")
-
-        return img
-
-    else:
-        image_url = (
-            f"https://api.brain-map.org/api/v2/section_image_download/{image_id}"
-        )
-        options = []
-
+    folder = pathlib.Path(folder)
+    folder.mkdir(exist_ok=True, parents=True)
+
+    # Construct the image file name and the full path
+    file_name = f"{image_id}-{downsample}"
+    if expression:
+        file_name += "-expression"
+    image_path = folder / (file_name + ".jpg")
+
+    # Download the image if not already in the cache
+    if not image_path.exists():
+        base_url = "https://api.brain-map.org/api/v2/section_image_download"
+        url = f"{base_url}/{image_id}?downsample={downsample}"
         if expression:
-            options.append("view=expression")
-
-        if downsample:
-            options.append(f"downsample={downsample}")
+            url += "&view=expression"
 
-        image_url += "?" + "&".join(options)
+        # Download the image
+        response = requests.get(url)
+        response.raise_for_status()
+        with image_path.open("wb") as fp:
+            fp.write(response.content)
 
-        response = requests.get(image_url, stream=True)
-        if not response.ok:
-            raise ValueError("Request failed!")
+    # Read the cached image from disk
+    with Image.open(image_path) as lazy_img:
```

Comment:
As opposed to writing (see my comment above) here we use a 3rd party package. I guess there are no ways around it, or are there? I guess we introduce the `Pillow` dependency just because of this line. Maybe we can use a dependency that we already depend on?

Reply:
Yeah, so like I wrote above, it's all about converting the JPEG bytes to NumPy arrays. I don't think we can do that without a library. And we can't avoid dealing with JPEG because that's what AIBS sends to us.

Regarding extra dependencies, formally there's a new "Pillow" entry in our requirements, but in practise this package will always be installed as part of the `matplotlib` requirements anyway. Actually, `plt.imread` just uses `PIL` under the hood anyway, so the logic here was to avoid the heavy `matplotlib` import and just go directly for `PIL`.

Reply:
Great points!